### ISSUE TITLE
log_json prepends logs with whitespace

### ISSUE THREAD

**Issue by gmichalec-pandora** [hashicorp/nomad#9489]
### Nomad version
Nomad v0.12.5+ent

### Operating system and Environment details
Debian Buster

### Issue
When enabling `log_json = true` in the server or client configs, it looks like the log output has a number of spaces at the beginning.
```
$ sudo journalctl -r -u nomad
-- Logs begin at Mon 2020-11-30 18:15:04 PST, end at Tue 2020-12-01 15:05:08 PST. --
Dec 01 15:05:02 dc6-docker3 nomad[28947]:     {"@level":"debug","@message":"request complete","@module":"http","@timestamp":"2020-12-01T15:05:02.049190-08:00","duration":14194187,"method":"GET","path":"/v1/node/df64379a-2fa8-811c-5656-bdf6268428e7/allocations"}
Dec 01 15:05:02 dc6-docker3 nomad[28947]:     {"@level":"debug","@message":"request complete","@module":"http","@timestamp":"2020-12-01T15:05:02.030335-08:00","duration":787602,"method":"GET","path":"/v1/agent/self"}
Dec 01 15:05:01 dc6-docker3 nomad[28947]:     {"@level":"debug","@message":"request complete","@module":"http","@timestamp":"2020-12-01T15:05:01.876159-08:00","duration":679382,"method":"GET","path":"/v1/node/df64379a-2fa8-811c-5656-bdf6268428e7"}
Dec 01 15:05:01 dc6-docker3 nomad[28947]:     {"@level":"debug","@message":"request complete","@module":"http","@timestamp":"2020-12-01T15:05:01.864813-08:00","duration":599042,"method":"GET","path":"/v1/agent/self"}

```
This requires extra config on our log ingester (logstash) in order to parse the JSON, as it assumes a JSON log begins with "{". Not a huge issue, but I can't think of a reason why one would ever want extra whitespace in a JSON-formatted log

**Comment by tgross** [comment]
Hi @gmichalec-pandora! Thanks for reporting this. It looks like a bug somewhere in the setup of the logger in [`command.go`](https://github.com/hashicorp/nomad/blob/v1.0.0-beta3/command/agent/command.go#L607-L640). We'll take a look, but if you're up for submitting a PR for a fix I'd be happy to review as well.

**Comment by tgross** [comment]
Closed by #9795 and this will ship in the next standard patch release.
